src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -236,6 +296,47 @@ impl<'tcx> Loan<'tcx> {
     pub fn loan_path(&self) -> Rc<LoanPath<'tcx>> {
         self.loan_path.clone()
     }
+
+    pub fn kind(&self) -> ty::BorrowKind {
+        self.kind
+    }
+
+    pub fn span(&self) -> Span {
+        self.span
+    }
+
+    pub fn span_in_def(&self, def_id: DefId, tcx: TyCtxt) -> Option<Span> {
+        let fn_span = tcx.hir
+            .as_local_node_id(def_id)
+            .map(|node_id| tcx.hir.span(node_id));
+
+        let gen_span = self.gen_scope.span(&tcx.hir)
+            .and_then(|s| fn_span.and_then(|n| Self::get_unexpanded_span(s, n)));
+
+        let kill_span = self.kill_scope.span(&tcx.hir)
+            .and_then(|s| fn_span.and_then(|n| Self::get_unexpanded_span(s, n)));
+
+        match (gen_span, kill_span) {
+            (Some(gen_span), Some(kill_span)) => {
+                Some(Span {
+                    lo: gen_span.lo,
+                    hi: kill_span.hi,
+                    ctxt: NO_EXPANSION
+                })
+            },
+            _ => None,
+        }
+    }
+
+    fn get_unexpanded_span(input_span: Span, wrapping_span: Span) -> Option<Span> {
```

Comment:
The answer to a question like this is always "add a comment". =) That would not have been obvious to me. I think an example of some Rust code where this is relevant would be even better.